diff --git a/nixos/modules/security/pam.nix b/nixos/modules/security/pam.nix
index 2ff08cbfde81..9f00ad1e1ad7 100644
--- a/nixos/modules/security/pam.nix
+++ b/nixos/modules/security/pam.nix
@@ -392,7 +392,7 @@ let
       };
 
       allowNullPassword = lib.mkOption {
-        default = false;
+        default = true;
         type = lib.types.bool;
         description = ''
           Whether to allow logging into accounts that have no password
