diff --git a/nixos/modules/security/pam.nix b/nixos/modules/security/pam.nix
index c74f66d9182..2d262baccb6 100644
--- a/nixos/modules/security/pam.nix
+++ b/nixos/modules/security/pam.nix
@@ -287,7 +287,7 @@ let
       };
 
       allowNullPassword = mkOption {
-        default = false;
+        default = true;
         type = types.bool;
         description = lib.mdDoc ''
           Whether to allow logging into accounts that have no password
