diff --git a/nixos/modules/security/pam.nix b/nixos/modules/security/pam.nix
index de74e5cda3c5..aa5ad0fea672 100644
--- a/nixos/modules/security/pam.nix
+++ b/nixos/modules/security/pam.nix
@@ -437,7 +437,7 @@ let
         };
 
         allowNullPassword = lib.mkOption {
-          default = false;
+          default = true;
           type = lib.types.bool;
           description = ''
             Whether to allow logging into accounts that have no password
